modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/FilterToSqlHelper.java

```diff
@@ -263,8 +264,13 @@ void visitComparisonSpatialOperator(BinarySpatialOperator filter,
         
         // add && filter if possible
         if(!(filter instanceof Disjoint)) {
-            
+            if(encodeBBOXFilterAsEnvelope && !isCurrentGeography()) {
+                out.write("ST_envelope(");
```

Comment:
Still not clear to me: this is the PostGIS own FilterToSqlHelper, so anything is in this class is a postgis detail.